Assets/Scripts/navigation/Navigation.cs

```diff
@@ -0,0 +1,186 @@
+			Vector3 delta = from - to;
+			return Mathf.Atan2(delta.x, delta.z) * 180 / Mathf.PI;
+		}
+		public static float EulerNormalize(float angle) {
+			while (angle < -180) angle += 360;
+			while (angle > 180) angle -= 360;
+			return angle;
+		}
+		public void Setdestination(string target, int ID) {
+			if (target == null) {
+				Debug.LogError("Didn't find the city");
+				return;
+			}
+			Vector3 targetlocaion = cities.GetCityLocation(target);
+			if (targetlocaion == null) {
+				Debug.LogError("Unable to find city location");
+				return;
+			}
+			//assign image
+			ImageSlot.sprite = Resources.Load<Sprite>(ResourcePath + "/" + ID) ?? Resources.Load<Sprite>(DefaultPortrait);
+			StartNavigation = true;
+			//find player
+			player = GameObject.Find("playerShip").transform.Find("ShipParent").transform.Find("kyrenia_parent").transform;
+			this.transform.position = player.position;
+			float distance = Vector3.Distance(targetlocaion, player.position);
+			float radius = 2;
+			if (distance >= radius) {
+				string Saildirecation = FindPlayerDirection(player); // find what angle is player faceing
+				string Citydirecation = FindCityDirection(targetlocaion,player); // find city angle by the world
+				string WhereToSail = getAIPath(targetlocaion, radius); // find the path
+				text.text = Citydirecation +Saildirecation + WhereToSail; // assign text
+			}
+			if (distance < radius) {
+				text.text = "You have arrive to " + target + ".";
+			}
+		}
+		string FindCityDirection(Vector3 targetlocaion, Transform current) {
+			float angle = CalcAngle(current.position + -current.up, targetlocaion);    // target angle relative to world
+			angle -= 180;
+			string citydirecation = "The city is in the "+ FindDirection(angle)+".";
+			return citydirecation;
+		}
+		string FindPlayerDirection(Transform current) {
+			float angle2 = CalcAngle(current.position, current.position + current.up); // player angle relative to world
+			string Saildirecation = "The Ship is sail to the " + FindDirection(angle2) + ".";
+			return Saildirecation;
+		}
+		string getAIPath(Vector3 target,float radius) {
+			AI.stoppingDistance = radius;
+			AI.SetDestination(target);
+			Vector3[] path = AI.path.corners;
+			//lineRenderer.positionCount = AI.path.corners.Length;
+			if (path.Length <= 1) {
+				return "";
+			}
+			//for (int i = 0; i < path.Length; i++) {
+			//	lineRenderer.SetPosition(i, path[i]);
+			//}
+			//get next point
+			if(path.Length > 3) {
+				for (int i = 1; i < AI.path.corners.Length; i++) {
+					float dis = Vector3.Distance(player.position, AI.path.corners[i]);
+					if (dis < 6) {
+						int z = (i + 1) <= AI.path.corners.Length ? i + 1 : (AI.path.corners.Length - 1);
+						for (int j = i; j <= AI.path.corners.Length - 1; j++) {
+							var tmp = (j + 1) <= (AI.path.corners.Length - 1) ? Vector3.Distance(AI.path.corners[j], AI.path.corners[j + 1]) : Vector3.Distance(AI.path.corners[j], AI.path.corners[AI.path.corners.Length - 1]);
```

Comment:
make sure to use descriptive names for variables. the names z and tmp don't help communicate how the code works, and even you will probably forget after a year. can you do a pass on renaming? 

i think in this case i and j should be renamed as well because it's not quite clear what the second for loop is doing for instance, although in simpler code i do think it's okay to name iteration variables like this something simple like i and j if for ex. you're just looping through a 2d matrix. it's just in this case it's not clear what they mean